drivers/wifi/infineon/airoc_wifi.c

```diff
@@ -0,0 +1,652 @@
+	}
+	*buffer = buf;
+	return WHD_SUCCESS;
+}
+
+void cy_buffer_release(whd_buffer_t buffer, whd_buffer_dir_t direction)
+{
+	CY_UNUSED_PARAMETER(direction);
+	(void)net_buf_destroy((struct net_buf *)buffer);
+}
+
+uint8_t *cy_buffer_get_current_piece_data_pointer(whd_buffer_t buffer)
+{
+	CY_ASSERT(buffer != NULL);
+	struct net_buf *buf = (struct net_buf *)buffer;
+
+	return (uint8_t *)buf->data;
+}
+
+uint16_t cy_buffer_get_current_piece_size(whd_buffer_t buffer)
+{
+	CY_ASSERT(buffer != NULL);
+	struct net_buf *buf = (struct net_buf *)buffer;
+
+	return (uint16_t)buf->size;
+}
+
+whd_result_t cy_buffer_set_size(whd_buffer_t buffer, unsigned short size)
+{
+	CY_ASSERT(buffer != NULL);
+	struct net_buf *buf = (struct net_buf *)buffer;
+
+	buf->size = size;
+	return CY_RSLT_SUCCESS;
+}
+
+whd_result_t cy_buffer_add_remove_at_front(whd_buffer_t *buffer, int32_t add_remove_amount)
+{
+	CY_ASSERT(buffer != NULL);
+	struct net_buf **buf = (struct net_buf **)buffer;
+
+	if (add_remove_amount > 0) {
+		(*buf)->len = (*buf)->size;
+		(*buf)->data = net_buf_pull(*buf, add_remove_amount);
+	} else {
+		(*buf)->data = net_buf_push(*buf, -add_remove_amount);
+		(*buf)->len = (*buf)->size;
+	}
+	return WHD_SUCCESS;
+}
+
+static int airoc_mgmt_send(const struct device *dev, struct net_pkt *pkt)
+{
+	struct airoc_wifi_data *data = dev->data;
+	cy_rslt_t ret = CY_RSLT_SUCCESS;
+	size_t pkt_len = net_pkt_get_len(pkt);
+	struct net_buf *buf = NULL;
+
+	/* Allocate Network Buffer from pool with Packet Length + Data Header */
+	buf = net_buf_alloc_len(&airoc_pool, pkt_len + sizeof(data_header_t), K_NO_WAIT);
+	if (buf == NULL) {
+		return -EIO;
+	}
+
+	/* Reserve the buffer Headroom for WHD Data header */
+	net_buf_reserve(buf, sizeof(data_header_t));
+
+	/* Read the packet payload */
+	if (net_pkt_read(pkt, buf->data, pkt_len) < 0) {
+		LOG_ERR("net_pkt_read failed");
+		return -EIO;
+	}
+
+	/* Call WHD API to send out the Packet */
+	ret = whd_network_send_ethernet_data(airoc_if, (void *)buf);
+	if (ret != CY_RSLT_SUCCESS) {
+		LOG_ERR("whd_network_send_ethernet_data failed");
+#if defined(CONFIG_NET_STATISTICS_WIFI)
+		data->stats.errors.tx++;
+#endif
+		return -EIO;
+	}
+
+#if defined(CONFIG_NET_STATISTICS_WIFI)
+	data->stats.bytes.sent += pkt_len;
+	data->stats.pkts.tx++;
+#endif
+
+	return 0;
+}
+
+void cy_network_process_ethernet_data(whd_interface_t interface, whd_buffer_t buffer)
+{
+	struct net_pkt *pkt;
+	uint8_t *data = whd_buffer_get_current_piece_data_pointer(interface->whd_driver, buffer);
+	uint32_t len = whd_buffer_get_current_piece_size(interface->whd_driver, buffer);
+	bool net_pkt_unref_flag = false;
+
+	if ((interface->role == WHD_STA_ROLE) &&
+	    (net_if_flag_is_set(airoc_wifi_iface, NET_IF_UP))) {
+		if (airoc_wifi_iface == NULL) {
+			LOG_ERR("network interface unavailable");
+			cy_buffer_release(buffer, WHD_NETWORK_RX);
+			return;
+		}
+
+		pkt = net_pkt_rx_alloc_with_buffer(airoc_wifi_iface, len, AF_UNSPEC, 0, K_NO_WAIT);
+
```

Comment:
Typically, NULL Checks don't need a newline as they are the same logical block.

Reply:
updated